### PR TITLE
Update geth version for integration tests

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
- update geth version for integration tests from `1.8.1` to `1.8.22` (inspired by [convo here](https://github.com/ethereum/web3.py/pull/1295#issuecomment-475323375))

- update go version used in integration tests

- removed unused geth 1.6 fixture data

- xfail test for `eth_submitHashrate` in `1.8.22` geth tests as it appears it has been dropped in favor of `ethash_submitHashRate`

#### Cute Animal Picture
image

### CODE REVIEW

File: tests/integration/go_ethereum/common.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
     Web3ModuleTest,
 )
 
-GETH_SIGNED_TX = b'\xf8j\x80\x85\x040\xe24\x00\x82R\x08\x94\xdcTM\x1a\xa8\x8f\xf8\xbb\xd2\xf2\xae\xc7T\xb1\xf1\xe9\x9e\x18\x12\xfd\x01\x80\x86\xee\xca\xc4f\xe1\x16\xa0\xbb7^\x1f\xf0\x03(P\x07|\x053Q\xd3M\xf1\x83\xe9\xdcp\xdc\x02\xb4\xe7`\x85\xcd\x84\xdb\xb4\xd0\xaa\xa07\x8cl\xd7\xa6R\x01\xfaW\x0e\x0f\xc1_$\xdf`\x8dO\x18\x1dC\xbc\x87\x8fud\xd2R*W\xfd4'  # noqa: E501
+GETH_17_SIGNED_TX = b'\xf8j\x80\x85\x040\xe24\x00\x82R\x08\x94\xdcTM\x1a\xa8\x8f\xf8\xbb\xd2\xf2\xae\xc7T\xb1\xf1\xe9\x9e\x18\x12\xfd\x01\x80\x86\xee\xca\xc4f\xe1\x16\xa0\xbb7^\x1f\xf0\x03(P\x07|\x053Q\xd3M\xf1\x83\xe9\xdcp\xdc\x02\xb4\xe7`\x85\xcd\x84\xdb\xb4\xd0\xaa\xa07\x8cl\xd7\xa6R\x01\xfaW\x0e\x0f\xc1_$\xdf`\x8dO\x18\x1dC\xbc\x87\x8fud\xd2R*W\xfd4'  # noqa: E501
+GETH_18_SIGNED_TX = b'\xf8i\x80\x84;\x9a\xca\x00\x82R\x08\x94\xdcTM\x1a\xa8\x8f\xf8\xbb\xd2\xf2\xae\xc7T\xb1\xf1\xe9\x9e\x18\x12\xfd\x01\x80\x86\xee\xca\xc4f\xe1\x16\xa0W\x91\xe6a\xb7l\x93,\xd3\xdd?;\xa8\x0e\xd3\xbe\x04\x89(\xd49\x1e+\xd6\xee\x88k\xfb\xe7\x83\xeb(\xa0\x04\x9a%`\x91uc\x1a\xdd\xb8\x96\xbb\x10\xb5v|\xcf\x03\xb6\x90\xbb\x93x\xfef\xae\xe1L\xbc7\xafJ'  # noqa: E501
```
Comment (njgheorghita) [inline tests/integration/go_ethereum/common.py]: I'm having a hard time figuring out why these are returning different values for a signed tx. If the parameters are the same (i.e. gas price, to / from, tx...) shouldn't the signed transaction be the same? Why could different versions of geth be returning different values?

File: tests/integration/go_ethereum/common.py [hunk 2]
```diff
@@ -62,7 +63,16 @@ def test_eth_estimateGas_with_block(self,
...
+        if 'v1.8.22' in web3.clientVersion:
+            super().test_eth_signTransaction(web3, unlocked_account, GETH_18_SIGNED_TX)
+        else:
+            super().test_eth_signTransaction(web3, unlocked_account, GETH_17_SIGNED_TX)
+
+    def test_eth_submitHashrate(self, web3):
+        if 'v1.8.22' in web3.clientVersion:
+            # https://github.com/ethereum/go-ethereum/commit/51db5975cc5fb88db6a0dba1826b534fd4df29d7
+            pytest.xfail('eth_submitHashrate endpoint updated in 1.8.22 to ethash_submitHashRate')
+        super().test_eth_submitHashrate(web3)
```
Comment (njgheorghita) [inline tests/integration/go_ethereum/common.py]: From what I can tell - geth no longer supports the `eth_submitHashrate` endpoint. What's the strategy for `web3.eth.submitHashrate`? Better to just leave it alone and let users running the latest geth run into `"this method is not available"` errors?

File: tests/integration/go_ethereum/common.py (hunk 1 above)
Comment (pipermerriam) [reply tests/integration/go_ethereum/common.py]: I'd decode it (`rlp.decode(...)`) and see what is different.

**Comment by njgheorghita** [comment]
@pipermerriam ping for final review

**Pull request merged**